Submission.java

```diff
@@ -181,23 +258,95 @@ private JSONArray getSprites()
      * Get script count for sprite.
      *
      * @param spriteName 
-     * @return scriptCount 
+     * @return count 
      */
     public int getScriptCountForSprite(String spriteName)
     {
+        return getCountForSprite("scripts", spriteName);
+    }
+    
+    /**
+     * Get variable count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getVariableCountForSprite(String spriteName)
+    {
+        return getCountForSprite("variables", spriteName);
+    }
+    
+    /**
+     * Get list count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getListCountForSprite(String spriteName)
+    {
+        return getCountForSprite("lists", spriteName);
+    }
+    
+    /**
+     * Get script comment count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getScriptCommentCountForSprite(String spriteName)
+    {
+        return getCountForSprite("scriptComments", spriteName);
+    }
+
+    /**
+     * Get sound count for sprite.
+     *
+     * @param spriteName 
+     * @return count
+     */
+    public int getSoundCountForSprite(String spriteName)
+    {
+        return getCountForSprite("sounds", spriteName);
+    }
+
+    /**
+     * Get costume count for sprite.
+     *
+     * @param spriteName 
+     * @return count
+     */
+    public int getCostumeCountForSprite(String spriteName)
+    {
+        return getCountForSprite("costumes", spriteName);
+    }
+    
+    /**
+     * Get count for sprite.
+     *
+     * @param item 
+     * @param spriteName 
+     * @return count 
+     */
+    private int getCountForSprite(String item, String spriteName)
+    {
         JSONArray sprites = getSprites();
-        JSONArray scripts = new JSONArray();
+        JSONArray items = new JSONArray();
         for (int i = 0; i < sprites.size(); i++)
```

Comment:
@haietza That makes sense.   Maybe you could incorporate that into the method's javadoc comment, something like "this is a generic helper function used by the countForSprite methods"?  Idk, what do you think?